mllib/src/main/scala/org/apache/spark/ml/classification/NaiveBayes.scala

```diff
@@ -129,29 +129,49 @@ class NaiveBayesModel private[ml] (
       throw new UnknownError(s"Invalid modelType: ${$(modelType)}.")
   }
 
-  override protected def predict(features: Vector): Double = {
+  override val numClasses: Int = pi.size
+
+  private def posteriorProbabilities(logProb: DenseVector) = {
```

Comment:
Yes, posteriorProbabilities is easy to reuse, but  it not easy to directly reuse multinomialCalculation, and bernoulliCalculation because the mllib.NaiveBayesModel and ml.NaiveBayesModel has different model parameters.
```java
class NaiveBayesModel private[ml] (
    override val uid: String,
    val pi: Vector,
    val theta: Matrix)
```